spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java

```diff
@@ -95,4 +97,17 @@ public void destroyAll() {
 		restTemplate.delete(resources.get("streams/definitions").expand());
 	}
 
+	@Override
+	public void deleteQueue(String name, Map<String, String> properties) {
+		String rootUri = resources.get("streams/clean/rabbit").toString() + "/{name}";
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(rootUri);
+
+		for(Map.Entry<String, String> entry: properties.entrySet()) {
+			builder.queryParam(entry.getKey(), entry.getValue());
+		}
```

Comment:
> Fine with me (intermediate class); I simply abhor seeing essentially identical code in sibling classes - it's a maintenance nightmare.

Just to confirm with you @garyrussell, you'd like to see us break [the pattern set](https://github.com/spring-projects/spring-xd/blob/master/spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java#L72-L96) by the [rest of the code](https://github.com/spring-projects/spring-xd/blob/master/spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/JobTemplate.java#L111-L133) in the templates?